packages/editor/src/components/rich-text/index.js

```diff
@@ -941,7 +953,7 @@ const RichTextContainer = compose( [
 
 		// Ensures that only one RichText component can be focused.
 		return {
-			isSelected: context.isSelected && context.focusedElement === ownProps.instanceId,
+			isSelected: context.isSelected,
```

Comment:
@gziolo You introduced this change in ead8dc5b46f3c801122d925ee9a4019cb7a40ef8. In what scenario can more than one RichText instance be focussed?

Reply:
Additionally, would this PR introduce any bugs #6419 fixed with this line change, and if so, what are the steps to reproduce. I wonder then if there's no other way.

Reply:
> In what scenario can more than one RichText instance be focussed?

Only one `RichText` can be focused at the time. This code ensures that formatting toolbars are rendered only for the currently active `RichText`. It was buggy before and you could see multiple controls of the same type for the block that are using multiple `RichText` instances when `isSelected` wasn't properly handled outside. If you remove this code you need to ensure, that formatting controls are only rendered for the `RichText` you are actually using. The easiest way to reproduce is to put two or more `RichText` components into a block and don't pass `isSelected` prop.

If we remove `setFocusedElement`, we should also remove it from a few other places as it becomes dead code.

Reply:
I think the biggest challenge to implement it inside `RichText` is that it's very to control it through `focus` and `blur` events given that formatting controls are rendered outside of the `RichText` component's virtual DOM. It might be different now so it's something worth exploring. This `setFocusedElement` isn't ideal. It was introduced to make developer's life easier. We had multiple reports of seeing duplicated controls, including Columns block in Gutenberg :)

Reply:
@gziolo This seems to be still an issue in `master`:

![gallery](https://user-images.githubusercontent.com/4710635/47901514-74b50280-de80-11e8-9051-e1185e468b79.gif)

Can you reproduce?

Reply:
Gallery handles `RichText` on its own as its opt in for it by using `isSelected`. See:
https://github.com/WordPress/gutenberg/blob/master/packages/block-library/src/gallery/gallery-image.js#L132

I think this is a regression from either refactoring Gallery block or RichText component